src/ui/Analytics/Analytics.ts

```diff
@@ -371,6 +373,42 @@ export class Analytics extends Component {
     this.client.setOriginContext(originContext);
   }
 
+  /**
+   * Re-enables analytics if they were previously disabled.
+   */
+  public enable() {
+    if (!this.disabled) {
+      console.warn('Analytics are already enabled.');
+      return;
+    }
+    super.enable();
+    this.initializeAnalyticsClient();
+    this.resolveQueryController().historyStore = new HistoryStore();
+  }
+
+  /**
+   * Removes the analytics tracking cookie and clears the actions history.
+   */
+  public clearLocalData() {
+    this.client.endpoint.forgetVisitorId();
+    this.resolveQueryController().historyStore.clear();
+  }
+
+  /**
+   * Disables analytics and clears local data by running [`clearLocalData`]{@link Analytics.clearLocalData}.
+   */
+  public disable() {
+    if (this.disabled) {
+      console.warn('Analytics are already disabled.');
+      return;
+    }
+    super.disable();
+    this.clearLocalData();
+    this.client.endpoint.kill();
+    this.client = new NoopAnalyticsClient();
+    this.resolveQueryController().historyStore = new HistoryStore(new NullStorage());
```

Comment:
Similar comment about moving this to the `QueryController` class.

Reply:
Yes, such manipulations should be done in an abstract way, and triggered by exposing a public method such as `.resetHistory()`